platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
+    fn metadata(resource_id: ResourceId) -> Result<Metadata, String>;
+    fn read(resource_id: ResourceId) -> Result<Content, String>;
+    fn remove(resource_id: ResourceId) -> Result<(), String>;
+    // import will decide if it needs to unzip a file. The contents of the url will be extracted and inserted into a database.
+    fn import(
+        url: String,
```

Comment:
This isn't necessarily a URL - this is just how the host operating system denotes local files. Android has URLs, IIRC iOS has paths, but we don't really know what it's going to be, because that depends on the OS. Not sure what to call it, `reference`? `localReference`? `localHandle`? Dunno, but `url` seems wrong :D

Reply:
That'd imply that there may not be a path structure in there - which we want to expose to the user, this being a virtual file system. A URN for example is a valid URI, but not a URL.

Reply:
We can force the platforms to give us this as an url. 

To me an url makes most sense here. Can we have a case where a feature tries to import an archive from dropbox or smth like that?